src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.FullGlobalizationData.Unix.cs

```diff
@@ -256,8 +288,14 @@ private static string GetExemplarCityName(string timeZoneId, string uiCultureNam
         // Helper function that returns an alternative ID using ICU data. Used primarily for converting from Windows IDs.
         private static unsafe string? GetAlternativeId(string id, out bool idIsIana)
         {
+#if TARGET_MACCATALYST || TARGET_IOS || TARGET_TVOS
+            // No alternative IDs in this target.
```

Comment:
I noticed that this isn't conditioned on `GlobalizationMode.Hybrid` whereas other logic originating from `TimeZoneInfo.MinimalGlobalizationData.cs` are
i.e. 
```
#if TARGET_MACCATALYST || TARGET_IOS || TARGET_TVOS
            if (!GlobalizationMode.Hybrid)
            {
                // For this target, be consistent with other time zone display names that use an abbreviation.
                return "UTC";
            }
#endif
```
and 
```
#if TARGET_MACCATALYST || TARGET_IOS || TARGET_TVOS
            if (!GlobalizationMode.Hybrid)
            {
                // For this target, be consistent with other time zone display names that use the ID.
                return $"(UTC) {timeZoneId}";
            }
#endif
```
in `TimeZoneInfo.FullGlobalizationData.Unix.cs` additions.

Why is that the case? This logic is also unnecessary as we have it in `TryConvertWindowsIdToIanaId` itself.

Reply:
`TryConvertWindowsIdToIanaId` and `TryConvertWindowsIdToIanaId` are not conditioned as I couldn't find equivalents in Apple docs, so for these cases I kept old logic (as old logic is also not using ICU).

Reply:
Ah, that makes sense. This logic is already covered by the 
```
#if TARGET_MACCATALYST || TARGET_IOS || TARGET_TVOS
            ianaId = null;
            return false;
#else
```
in `TimeZoneInfo.FullGlobalizationData.cs` so we dont need to change `GetAlternativeId` at all

Reply:
Good catch, will remove from here.